nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -493,6 +502,7 @@ void RegulatedPurePursuitController::reset()
 {
   cancelling_ = false;
   finished_cancelling_ = false;
+  has_reached_xy_tolerance_ = false;
```

Comment:
I think it also needs to be reset when a new plan is given in `setPlan`

Reply:
Done